server/src/main/java/org/candlepin/dto/api/v1/AsyncJobStatusTranslator.java

```diff
@@ -72,14 +71,54 @@ public AsyncJobStatusDTO populate(ModelTranslator translator, AsyncJobStatus sou
         destination.setAttempts(source.getAttempts());
         destination.setMaxAttempts(source.getMaxAttempts());
         destination.setResult(source.getJobResult());
+        destination.setState(stateToString(mapState(source.getState())));
+        destination.setPreviousState(stateToString(mapState(source.getPreviousState())));
 
-        JobState state = source.getState();
-        destination.setState(state != null ? state.name() : null);
+        return destination;
+    }
 
-        JobState pstate = source.getPreviousState();
-        destination.setPreviousState(pstate != null ? pstate.name() : null);
+    private String stateToString(PublicJobState state) {
+        if (state == null) {
+            return null;
+        }
+        return state.name();
+    }
 
-        return destination;
+    public static PublicJobState mapState(JobState state) {
+        if (state == null) {
+            return null;
+        }
+        PublicJobState publicJobState;
+        switch (state) {
+            case FINISHED:
+                publicJobState = PublicJobState.FINISHED;
+                break;
+            case CREATED:
+                publicJobState = PublicJobState.CREATED;
+                break;
+            case QUEUED:
```

Comment:
Should QUEUED be part of RUNNING or CREATED? This is another one where I can see arguments for either case.

Reply:
I guess you are right. Queued job is not running yet so we can merge the state into the CREATED instead of RUNNING. Done.